src/uct/ib/dc/dc_mlx5.c

```diff
@@ -684,7 +689,12 @@ uct_dc_mlx5_init_rx(uct_rc_iface_t *rc_iface,
     }
 
     if (iface->super.config.srq_topo == UCT_RC_MLX5_SRQ_TOPO_LIST) {
-        iface->super.super.progress = uct_dc_mlx5_iface_progress_ll;
+        if (iface->super.cq[UCT_IB_DIR_RX].zip ||
```

Comment:
So we check the same flags twice? First here, then in `uct_rc_mlx5_check_zip_flag`?

Reply:
If I understand you correctly - then the answer is here https://github.com/openucx/ucx/pull/8676#discussion_r1029999639